### ISSUE TITLE
[BUG] UNIX Redis socket: Celery expects different URI scheme "redis+socket" than application server

### ISSUE THREAD

**Issue by eomanis** [paperless-ngx/paperless-ngx#2108]
### Description

Between 1.9.2 and 1.10.1 paperless-ngx switched to Celery for its background scheduler, which has introduced a regression that prevents one from using a UNIX socket to communicate with Redis.

Tl;dr The paperless-ngx application server expects `unix:///path/to/redis.sock`, whereas Celery expects `redis+socket:///path/to/redis.sock`, but there is only one `PAPERLESS_REDIS=` setting for both of them.

If setting `PAPERLESS_REDIS=unix:///path/to/redis.sock`, Celery is unhappy:

```
celery beat v5.2.7 (dawn-chorus) is starting.
Traceback (most recent call last):
  File "/srv/paperless/python-venv/lib/python3.10/site-packages/kombu/transport/__init__.py", line 57, in resolve_transport
    transport = TRANSPORT_ALIASES[transport]
KeyError: 'unix'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/srv/paperless/python-venv/bin/celery", line 8, in <module>
    sys.exit(main())
  File "/srv/paperless/python-venv/lib/python3.10/site-packages/celery/__main__.py", line 15, in main
    sys.exit(_main())
  File "/srv/paperless/python-venv/lib/python3.10/site-packages/celery/bin/celery.py", line 217, in main
    return celery(auto_envvar_prefix="CELERY")
  File "/srv/paperless/python-venv/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
(...)
  File "/srv/paperless/python-venv/lib/python3.10/site-packages/kombu/connection.py", line 181, in __init__
    if not get_transport_cls(transport).can_parse_url:
  File "/srv/paperless/python-venv/lib/python3.10/site-packages/kombu/transport/__init__.py", line 85, in get_transport_cls
    _transport_cache[transport] = resolve_transport(transport)
  File "/srv/paperless/python-venv/lib/python3.10/site-packages/kombu/transport/__init__.py", line 66, in resolve_transport
    raise KeyError(f'No such transport: {transport}')
KeyError: 'No such transport: unix'
```

If setting `PAPERLESS_REDIS=redis+socket:///path/to/redis.sock`, Celery is satisfied, but the paperless-ngx application server complains:

```
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/srv/paperless/python-venv/lib/python3.10/site-packages/channels/utils.py", line 51, in await_many_dispatch
    await dispatch(result)
  File "/srv/paperless/python-venv/lib/python3.10/site-packages/asgiref/sync.py", line 435, in __call__
    ret = await asyncio.wait_for(future, timeout=None)
  File "/usr/lib/python3.10/asyncio/tasks.py", line 408, in wait_for
    return await fut
  File "/usr/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
(...)
  File "/srv/paperless/python-venv/lib/python3.10/site-packages/aioredis/commands/__init__.py", line 188, in create_redis_pool
    pool = await create_pool(address, db=db,
  File "/srv/paperless/python-venv/lib/python3.10/site-packages/aioredis/pool.py", line 39, in create_pool
    address, options = parse_url(address)
  File "/srv/paperless/python-venv/lib/python3.10/site-packages/aioredis/util.py", line 148, in parse_url
    assert r.scheme in ('', 'redis', 'rediss', 'unix'), (
AssertionError: ('Unsupported URI scheme', 'redis+socket')
```

My paperless.conf is this:
```
PAPERLESS_REDIS=unix:///srv/paperless/redis/redis.sock

PAPERLESS_CONSUMPTION_DIR=/srv/paperless/add
PAPERLESS_DATA_DIR=/srv/paperless/data
PAPERLESS_MEDIA_ROOT=/srv/paperless/media
PAPERLESS_STATICDIR=/srv/paperless/webroot/static

PAPERLESS_SECRET_KEY="XXXXXXXXXXXXXXXXXXXXXX"
PAPERLESS_URL=https://my-server:3333
PAPERLESS_STATIC_URL=/static/

PAPERLESS_OCR_LANGUAGE=deu
PAPERLESS_OCR_OUTPUT_TYPE=pdf
PAPERLESS_CONVERT_TMPDIR=/srv/paperless/tmp

PAPERLESS_SCRATCH_DIR=/srv/paperless/uploads

PAPERLESS_TASK_WORKERS=2
PAPERLESS_CONSUMER_POLLING=10
```

### Steps to reproduce

Have a dedicated Redis instance for paperless-ngx that hosts on a UNIX socket, and try to configure paperless-ngx to access this Redis instance through this UNIX socket.

### Webserver logs

```bash
See description
```

### Paperless-ngx version

1.10.1

### Host OS

Arch Linux x86_64

### Installation method

Bare metal

### Browser

Firefox 107.0.1

### Configuration changes

See description

**Comment by shamoon** [comment]
Duplicate of #2072 closed by #2092

**Comment by eomanis** [comment]
Damn you guys are quick, I gotta search the closed bugs too next time
